SYCL/ESIMD/regression/spirv_Function_Call_Test.cpp

```diff
@@ -0,0 +1,119 @@
+//==------- spirv_Function_Call_Test.cpp - DPC++ ESIMD on-device test-------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// This is a test to validate sycl-post-link correctly lowers calls to spirv
+// builtin globals
+
+#include "../esimd_test_utils.hpp"
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <cstdint>
+#include <iostream>
+
+using namespace sycl;
+using namespace sycl::ext::intel::esimd;
+
+SYCL_EXTERNAL inline size_t __spirv_GlobalInvocationId_x()
+#ifdef __SYCL_DEVICE_ONLY__
+    ;
+#else
+{
+  return 0;
+}
+#endif
+SYCL_EXTERNAL inline size_t __spirv_GlobalInvocationId_y()
+#ifdef __SYCL_DEVICE_ONLY__
+    ;
+#else
+{
+  return 0;
+}
+#endif
+SYCL_EXTERNAL inline size_t __spirv_GlobalInvocationId_z()
+#ifdef __SYCL_DEVICE_ONLY__
+    ;
+#else
+{
+  return 0;
+}
+#endif
+
+bool test(queue Q) {
+
+  constexpr unsigned Size = 1024;
+  constexpr unsigned VL = 32;
+  constexpr unsigned GroupSize = 8;
+
+  int *A = malloc_shared<int>(Size * sizeof(int), Q);
+  int *B = malloc_shared<int>(Size * sizeof(int), Q);
+  int *C = malloc_shared<int>(Size * sizeof(int), Q);
+
+  range<1> GlobalRange{Size / VL};
+  range<1> LocalRange{GroupSize};
+  nd_range<1> Range(GlobalRange, LocalRange);
+
+  try {
+    auto E = Q.submit([&](handler &CGH) {
+      CGH.parallel_for(Range, [=](nd_item<1> Ndi) SYCL_ESIMD_KERNEL {
+        using namespace sycl::ext::intel::esimd;
+        int I = Ndi.get_global_id(0);
+        for (auto Ind = 0; Ind < VL; ++Ind) {
+          A[I * VL + Ind] = __spirv_GlobalInvocationId_x();
+          B[I * VL + Ind] = __spirv_GlobalInvocationId_y();
+          C[I * VL + Ind] = __spirv_GlobalInvocationId_z();
```

Comment:
What __spirv_GlobalInvocationId_z() returns when the range is only 1 dimension?

Reply:
If it is 0 for _y, and 0 for _z, how do you know __spirv_GlobalInvocationId_z() is translated to something correct?, i.e. that it is not translated as if it was _y due to some error.

Please do not use internal functions. https://github.com/intel/llvm-test-suite/pull/1431#discussion_r1047947459